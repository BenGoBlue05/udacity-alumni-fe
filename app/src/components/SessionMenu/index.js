import React, { PropTypes } from 'react';
import Menu from 'grommet-udacity/components/Menu';
import Anchor from 'grommet-udacity/components/Anchor';
import Heading from 'grommet-udacity/components/Heading';
import Box from 'grommet-udacity/components/Box';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';

const NoUserMenu = ({
  pathname,
}) => (
  <Menu
    direction="row"
    align="center"
    inline
    resposive={false}
    className={styles.rightMenu}
  >
    <Anchor
      href="/login"
      className={pathname === '/login' ? 'active desktop__nav' : 'desktop__nav'}
    >
      Log In
    </Anchor>
    <Anchor
      href="/signup"
      className={pathname === '/signup' ? 'active desktop__nav' : 'desktop__nav'}
    >
      Sign Up
    </Anchor>
  </Menu>
);

const SessionIcon = ({
  user,
}) => (
  <Box
    responsive={false}
    direction="row"
    justify="center"
  >
    <img
      src={user.avatar}
      className={styles.userAvatar}
    />
    <Heading tag="h4" className={styles.profileName}>
      {user.name}
    </Heading>
  </Box>
);

const SessionMenu = ({
  onLogout,
  user,
  pathname,
}) => (
  <span>
    {(() => { // eslint-disable-line
      if (user) {
        switch (user.role) {
          case 'user':
            return (
              <Menu
                icon={<SessionIcon user={user} />}
                inline={false}
                dropAlign={{ bottom: 'bottom' }}
                a11yTitle="Session"
                className={styles.rightMenu}
              >
                <Anchor href="/me/profile">
                  My Profile
                </Anchor>
                <Anchor href="/logout" onClick={onLogout}>
                  Logout
                </Anchor>
              </Menu>
            );
          case 'admin':
            return (
              <Menu
                icon={<SessionIcon user={user} />}
                dropAlign={{ bottom: 'bottom' }}
                a11yTitle="Session"
                inline={false}
                className={styles.rightMenu}
              >
                <Anchor href="/me/profile">
                  My Profile
                </Anchor>
                <Anchor href="/admin/content-dashboard">
                  Dashboard
                </Anchor>
                <Anchor href="/admin/cms?action=new">
                  New Post
                </Anchor>
                <Anchor href="/logout" onClick={onLogout}>
                  Logout
                </Anchor>
              </Menu>
            );
          default:
            return (
              <NoUserMenu pathname={pathname} />
            );
        }
      } else {
        return (
          <NoUserMenu pathname={pathname} />
        );
      }
    })()}
  </span>
);

SessionMenu.propTypes = {
  user: PropTypes.object,
  pathname: PropTypes.string.isRequired,
};

export default cssModules(SessionMenu, styles);
