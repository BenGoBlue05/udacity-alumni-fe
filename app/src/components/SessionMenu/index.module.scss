.userAvatar {
  width: 48px;
  height: 48px;
  border-radius: 24px;
  border: 2px solid transparent;
  overflow: hidden;
  cursor: pointer;
  transition: all .3s ease-in-out;
}
