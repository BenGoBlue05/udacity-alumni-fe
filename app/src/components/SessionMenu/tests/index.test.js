import SessionMenu from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SessionMenu />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <SessionMenu
        user={{
          avatar: 'http://1onjea25cyhx3uvxgs4vu325.wpengine.netdna-cdn.com/wp-content/uploads/2016/05/image08.png',
        }}
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
