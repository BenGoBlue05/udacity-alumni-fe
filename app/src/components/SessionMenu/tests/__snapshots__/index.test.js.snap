exports[`<SessionMenu /> should render with default props 1`] = `
<Menu
  a11yTitle="Session"
  closeOnClick={true}
  direction="column"
  dropAlign={
    Object {
      "bottom": "bottom"
    }
  }
  icon={
    <function SessionIcon( /* istanbul ignore next */_ref) /* istanbul ignore next */{var
      user = _ref.user;return (
    
        /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Box2.default, /* istanbul ignore next */{
            responsive: false,
            direction: 'row',
            justify: 'center' },
    
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'img', /* istanbul ignore next */{
            src: user.avatar,
            className: styles.userAvatar }),
    
          /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_Heading2.default, /* istanbul ignore next */{ tag: 'h4', className: styles.profileName },
            user.name)));}
      user={
        Object {
          "avatar": "http://1onjea25cyhx3uvxgs4vu325.wpengine.netdna-cdn.com/wp-content/uploads/2016/05/image08.png"
        }
      } />
  }
  pad="none">
  <Anchor
    href="/profile"
    tag="a">
    Profile
  </Anchor>
  <Anchor
    href="/logout"
    tag="a">
    Logout
  </Anchor>
</Menu>
`;
